File: dentis-frontend/src/app/core/services/patient.service.ts
import { Injectable } from '@angular/core';
import { HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ApiService, PageResponse } from './api.service';
import { Patient, PatientSummary, CreatePatientRequest } from '../models/patient.model';

@Injectable({ providedIn: 'root' })
export class PatientService {
  constructor(private api: ApiService) {}

  create(request: CreatePatientRequest): Observable<Patient> {
    return this.api.post<Patient>('/patients', request);
  }

  update(id: string, request: CreatePatientRequest): Observable<Patient> {
    return this.api.put<Patient>(`/patients/${id}`, request);
  }

  findById(id: string): Observable<Patient> {
    return this.api.get<Patient>(`/patients/${id}`);
  }

  findAll(page = 0, size = 20): Observable<PageResponse<PatientSummary>> {
    const params = new HttpParams().set('page', page).set('size', size);
    return this.api.get<PageResponse<PatientSummary>>('/patients', params);
  }

  search(query: string, page = 0, size = 20): Observable<PageResponse<PatientSummary>> {
    const params = new HttpParams().set('q', query).set('page', page).set('size', size);
    return this.api.get<PageResponse<PatientSummary>>('/patients/search', params);
  }

  deactivate(id: string): Observable<void> {
    return this.api.delete<void>(`/patients/${id}`);
  }
}

File: dentis-application/src/main/java/com/dentis/application/service/PatientService.java
package com.dentis.application.service;

import com.dentis.application.dto.request.CreatePatientRequest;
import com.dentis.application.dto.request.UpdatePatientRequest;
import com.dentis.application.dto.response.PatientResponse;
import com.dentis.application.dto.response.PatientSummaryResponse;
import com.dentis.common.response.PageResponse;
import org.springframework.data.domain.Pageable;

public interface PatientService {

    PatientResponse create(CreatePatientRequest request);

    PatientResponse update(String id, UpdatePatientRequest request);

    PatientResponse findById(String id);

    PageResponse<PatientSummaryResponse> findAll(Pageable pageable);

    PageResponse<PatientSummaryResponse> search(String query, Pageable pageable);

    void deactivate(String id);
}
